eng/Build.props

```diff
@@ -160,6 +160,7 @@
         <DotNetProjects Include="
                           $(RepoRoot)src\Framework\App.Ref\src\Microsoft.AspNetCore.App.Ref.csproj;
                           $(RepoRoot)src\Framework\App.Ref.Internal\src\Microsoft.AspNetCore.App.Ref.Internal.csproj;
+                          $(RepoRoot)src\Framework\AspNetCoreAnalyzers\**\*.csproj;
```

Comment:
🤩

Reply:
@dotnet/aspnet-build @BrennanConroy This is already getting built and added to Microsoft.AspNetCore.App.Ref even though none of the tests run without it. I was able to find and fix a bug in DetectMisplacedLambdaAttribute.cs where it was emitting false positives by running the tests locally.

![ASP.NET Core refpack analyzers](https://user-images.githubusercontent.com/54385/191099937-661beb5e-dc3e-4536-96ee-e19ddc5fbb38.png)

What's going on here? Why is this getting published even though we weren't running the tests on the CI?

Reply:
Looks like it's built via https://github.com/dotnet/aspnetcore/blob/ce2db7ea0b161fc5eb35710fca6feeafeeac37bc/src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj#L72
and then bundled via 
https://github.com/dotnet/aspnetcore/blob/ce2db7ea0b161fc5eb35710fca6feeafeeac37bc/src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj#L176

Reply:
The tests don't run on this CI without the change. @BrennanConroy answered my question and explained why it was getting published before but not tested. Adding `$(RepoRoot)src\Framework\AspNetCoreAnalyzers\**\*.csproj;` to Build.props fixes the build so now the tests run on the CI. Can you please merge this as is?

Reply:
What tests❔ If there's a project or two that needs to reference the Analyzer or CodeFix project, that's where this should be fixed. Your change here only accidentally changes the build order and the redundancy will break things elsewhere.

In addition, any tests that run in Helix build the targeting pack and shared framework and send them to the Helix agents. The only case where it's possible to test w/o having built the App.Ref and App.Runtime projects is local testing on build agents and we don't do much of that anymore.

Reply:
```suggestion
                          $(RepoRoot)src\Framework\AspNetCoreAnalyzers\test\Microsoft.AspNetCore.App.Analyzers.Test.csproj;
```